php-checks/src/main/java/org/sonar/php/checks/SSLCertificatesVerificationDisabledCheck.java

```diff
@@ -0,0 +1,127 @@
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.php.checks;
+
+import com.google.common.collect.ImmutableSet;
+import java.util.List;
+import org.sonar.check.Rule;
+import org.sonar.php.checks.utils.CheckUtils;
+import org.sonar.php.tree.visitors.AssignmentExpressionVisitor;
+import org.sonar.plugins.php.api.symbols.Symbol;
+import org.sonar.plugins.php.api.tree.CompilationUnitTree;
+import org.sonar.plugins.php.api.tree.Tree;
+import org.sonar.plugins.php.api.tree.Tree.Kind;
+import org.sonar.plugins.php.api.tree.declaration.NamespaceNameTree;
+import org.sonar.plugins.php.api.tree.expression.ExpressionTree;
+import org.sonar.plugins.php.api.tree.expression.FunctionCallTree;
+import org.sonar.plugins.php.api.tree.expression.LiteralTree;
+import org.sonar.plugins.php.api.visitors.PHPVisitorCheck;
+
+@Rule(key = SSLCertificatesVerificationDisabledCheck.KEY)
+public class SSLCertificatesVerificationDisabledCheck extends PHPVisitorCheck {
+  public static final String KEY = "S4830";
+
+  private static final String CURL_SETOPT = "curl_setopt";
+  private static final String CURLOPT_SSL_VERIFYHOST = "CURLOPT_SSL_VERIFYHOST";
+  private static final String CURLOPT_SSL_VERIFYPEER = "CURLOPT_SSL_VERIFYPEER";
+
+  private static final String MESSAGE = "Change this code to enable trust chain verification.";
+  private AssignmentExpressionVisitor assignmentExpressionVisitor;
+
+  @Override
+  public void visitCompilationUnit(CompilationUnitTree tree) {
+    assignmentExpressionVisitor = new AssignmentExpressionVisitor(context().symbolTable());
+    tree.accept(assignmentExpressionVisitor);
+    super.visitCompilationUnit(tree);
+  }
+
+  @Override
+  public void visitFunctionCall(FunctionCallTree tree) {
+    String functionName = CheckUtils.getFunctionName(tree);
+    List<ExpressionTree> arguments = tree.arguments();
+
+    // Detect curl_setopt function usage
+    // http://php.net/manual/fr/function.curl-setopt.php
+    if (CURL_SETOPT.equals(functionName) && arguments.size() > 2) {
+      ExpressionTree optionArgument = arguments.get(1);
+      if (optionArgument.is(Tree.Kind.NAMESPACE_NAME) &&
+        CURLOPT_SSL_VERIFYHOST.equals(((NamespaceNameTree) optionArgument).name().text())) {
+        checkCURLSSLVerifyHost(getAssignedValue(arguments.get(2)));
+      } else if (optionArgument.is(Tree.Kind.NAMESPACE_NAME) &&
+        CURLOPT_SSL_VERIFYPEER.equals(((NamespaceNameTree) optionArgument).name().text())) {
+        checkCURLSSLVerifyPeer(getAssignedValue(arguments.get(2)));
+      }
+    }
+
+    // super method must be called in order to visit function call node's children
+    super.visitFunctionCall(tree);
+  }
+
+  private void checkCURLSSLVerifyHost(ExpressionTree curlOptValue) {
```

Comment:
there is some code duplicated between `checkCURLSSLVerifyHost` and `checkCURLSSLVerifyPeer`. Do you think you could improve that?